src/main/java/org/terasology/gookeeper/system/HungerSystem.java

```diff
@@ -0,0 +1,177 @@
+        if (event.getTarget().hasComponent(GooeyComponent.class)) {
+            EntityRef gooeyEntity = event.getTarget();
+
+            GooeyComponent gooeyComponent = gooeyEntity.getComponent(GooeyComponent.class);
+            HungerComponent hungerComponent = gooeyEntity.getComponent(HungerComponent.class);
+            HealthComponent healthComponent = gooeyEntity.getComponent(HealthComponent.class);
+
+            CharacterHeldItemComponent characterHeldItemComponent = event.getInstigator().getComponent(CharacterHeldItemComponent.class);
+
+            logger.info("Gooey Health: " + healthComponent.currentHealth);
+            if (characterHeldItemComponent != null && gooeyComponent.isCaptured) {
+                EntityRef item = characterHeldItemComponent.selectedItem;
+                String itemName = item.getComponent(DisplayNameComponent.class).name;
+
+                for (String acceptableFoodBlock : hungerComponent.foodBlockNames) {
+                    if (itemName.equals(acceptableFoodBlock)) {
+                        delayManager.cancelPeriodicAction(gooeyEntity, eventID);
+                        gooeyEntity.send(new GooeyFedEvent(event.getInstigator(), gooeyEntity, item));
+                        return;
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Receives GooeyFedEvent when the targeted gooey is fed the held food block and hence resets the health to max.
+     *
+     * @param event,entity   The GooeyFedEvent, the gooey entity
+     */
+    @ReceiveEvent(components = {GooeyComponent.class})
+    public void onGooeyFed(GooeyFedEvent event, EntityRef entityRef) {
+        HealthComponent healthComponent = event.getGooey().getComponent(HealthComponent.class);
+        HungerComponent hungerComponent = event.getGooey().getComponent(HungerComponent.class);
+
+        healthComponent.currentHealth = healthComponent.maxHealth;
+        delayManager.addPeriodicAction(event.getGooey(), eventID, hungerComponent.timeBeforeHungry, hungerComponent.healthDecreaseInterval);
+
+        event.getItem().destroy();
+        event.getGooey().saveComponent(healthComponent);
+    }
+
+    /**
+     * Receives PeriodicActionTriggeredEvent when the gooey entity's health is decreased
+     *
+     * @param event,entity   The PeriodicActionTriggeredEvent, the gooey entity
+     */
+    @ReceiveEvent(components = {GooeyComponent.class})
+    public void onGooeyHealthDecrease(PeriodicActionTriggeredEvent event, EntityRef entity) {
+        if (event.getActionId().equals(eventID)) {
+            HealthComponent healthComponent = entity.getComponent(HealthComponent.class);
+            HungerComponent hungerComponent = entity.getComponent(HungerComponent.class);
+
+            healthComponent.currentHealth -= hungerComponent.healthDecreaseAmount;
+
+            if (healthComponent.currentHealth <= 0) {
+                entity.destroy();
+            }
+            entity.saveComponent(healthComponent);
```

Comment:
Interesting,  this does not mess up the entity about to be destroyed if ```entity.destroy()``` is reached? I dont think I have every tried that.